src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -365,11 +365,12 @@ private string GetDefaultAuthorityInstance()
         /// <param name="authorityUri">Uri of the authority</param>
         /// <param name="isDefaultAuthority">Boolean telling if this is the default authority
         /// for the application</param>
+        /// <param name="validateAuthority">TODO(migration): documentation</param>
         /// <remarks>You can add several authorities, but only one can be the default authority</remarks>
         /// <returns>The builder to chain the .With methods</returns>
-        public T AddKnownAuthority(Uri authorityUri, bool isDefaultAuthority)
+        public T AddKnownAuthority(Uri authorityUri, bool isDefaultAuthority, bool validateAuthority = false)
```

Comment:
@MarkZuber, what does it do?

Reply:
If ValidateAuthority is not set, and the authority host is not in our static trusted host list, then we will get the AAD tenant discovery endpoint URL from the metadata server.  Otherwise, we will append v2.0/.well-known/openid-configuration to the canonical authority URL which is calculated from the discovery aliases.

public async Task<string> GetOpenIdConfigurationEndpointAsync(
            AuthorityInfo authorityInfo,
            string userPrincipalName,
            RequestContext requestContext)
        {
            var authorityUri = new Uri(authorityInfo.CanonicalAuthority);
            if (authorityInfo.ValidateAuthority && !AadAuthority.IsInTrustedHostList(authorityUri.Host))
            {
                var discoveryResponse = await _serviceBundle.AadInstanceDiscovery.GetMetadataEntryAsync(
                                            authorityUri,
                                            requestContext).ConfigureAwait(false);

                return discoveryResponse.TenantDiscoveryEndpoint;
            }

            return authorityInfo.CanonicalAuthority + Constants.OpenIdConfigurationEndpoint;
        }